mock-core-configs/mock-core-configs.spec

```diff
@@ -49,7 +49,15 @@ Config files which allow you to create chroots for:
 
 
 %build
-# nothing to do here
+
+%if 0%{?rhel} > 0 && 0%{?rhel} < 8
+# Required for dnf enabled ocnfigs on yum based hosts
+# config_opts['use_bootstrap_container'] = True
+grep -rl "config_opts\['package_manager'\] = 'dnf'" etc/mock | \
+    while read name; do
+    sed -i.bak "s/config_opts\['package_manager'\] = 'dnf'/config_opts\['package_manager'\] = 'dnf'\n# Enable bootstrap on dnf host for %%rhel %{rhel}\nconfig_opts\[\'use_bootstrap_container\'\] = True\n\n/g" $name
+done
```

Comment:
I dislike this;  I don't think it is wise to change the configuration only for RHEL.  Both dnf may be installed there, and users might only want to build in RHEL <= 7 chroots on RHEL 7.  And still, it is config only - we can rely on users to adjust the configuration accordingly.

If anything, we should have host-specific configuration, and only for those configuration files where this is absolutely necessary (fedora, rhel8+, ...)

Reply:
But the problem is not host specific. It is distribution specific. It applies only to specific operating systems which lack a functional native dnf (which I've so far only aidentified as RHEL based systems before RHEL 8), and only certain target configuration files (those which use on dnf). It's therefore not appropriate to site-defaults.cfg. or any equivalent.

If there is some graceful "if config_opts['package_manager'] = dnf" logic we can embed in these .cfg files, great! But in the meantime, this logic works well.

Reply:
We have a functional DNF in RHEL 7. Mock already isn't being updated in RHEL 6, so it's irrelevant for this matter.

Reply:
The issue with RHEL 7 hosts isn't DNF, but RPM. All currently supported Fedora releases and RHEL 8 use RPM features that don't exist in RHEL 7, which breaks your ability to create buildroots of those distributions.

For example, Mageia 6 and Mageia 7 buildroots work just fine on RHEL 7 because those features aren't in use in the core, permitting bootstrap chroots to work.

Reply:
I'd thought that those failures were tied to dnf, not RPM itself. If so, can you think of any better way to tetect what the local RPM  version is set to versus the alternative explanations? Outguessing which "cores" have, use or do not use such features sounds like a losing prospect.